sysl2/sysl/exporter/type_exporter.go

```diff
@@ -51,63 +51,58 @@ func (t *TypeExporter) populateTypes(syslTypes map[string]*proto.Type, swaggerTy
 			t.log.Warnf("Type matching failed %s", err)
 			return err
 		}
-		var ok bool
-		typeSchema.Format, ok = valueMap["format"]
-		typeSchema.Type = append(typeSchema.Type, valueMap["type"])
-		if !ok {
-			return fmt.Errorf("malformed sysl missing type info")
-		}
+		typeSchema.Format = valueMap.Format
+		typeSchema.Type = append(typeSchema.Type, valueMap.Type)
 		memberTypes := map[string]*proto.Type{}
 		if typeSchema.Properties == nil {
 			typeSchema.Properties = map[string]spec.Schema{}
 		}
-		if valueMap["format"] == "tuple" {
+		if valueMap.Format == "tuple" {
 			memberTypes = dataType.GetTuple().GetAttrDefs()
-		} else if valueMap["format"] == "relation" {
+		} else if valueMap.Format == "relation" {
 			memberTypes = dataType.GetRelation().GetAttrDefs()
 		}
-		var attrValueMap map[string]string
 		for attK, attV := range memberTypes {
 			elementSchema := spec.Schema{}
 			if t.isComposite(attV) {
 				t.parseComposite(attV, &elementSchema)
 				swaggerTypes[attK] = elementSchema
 			} else {
-				attrValueMap, err = t.findSwaggerType(attV)
+				attrValueMap, err := t.findSwaggerType(attV)
 				if err != nil {
 					t.log.Warnf("Type matching failed %s", err)
 					return err
 				}
+				elementSchema.Format = attrValueMap.Format
+				elementSchema.Type = []string{attrValueMap.Type}
 			}
-			elementSchema.Format = attrValueMap["Format"]
```

Comment:
@springwiz this is why maps are bad...

Reply:
@anz-gordonj7 You mean missing type info in yml is due to maps being used?

Reply:
no, the key is incorrect. "**f**ormat" was used everywhere else, but this one used "**F**ormat"

Reply:
That is a "typo", I could have used constants. It still doesn't prove maps are bad.

Reply:
Thanks for fixing :-)